### PR TITLE
convo 3: add/leave/update/updateKey

### PR DESCRIPTION
**Pull request by ianwremmel**
Relates to #63

### CODE REVIEW

**Comment by mwegman** [comment]
LGTM

File: packages/plugin-conversation/src/conversation.js [hunk 1]
```diff
@@ -101,6 +124,28 @@ const Conversation = SparkPlugin.extend({
       .then(() => res.body));
   },
 
+  leave(conversation, object, activity) {
+    this._inferConversationUrl(conversation);
```
Comment (Josh-Dykstra) [inline packages/plugin-conversation/src/conversation.js]: It seems like most places we handle the promise `_inferConversationUrl()` returns and treat it as though it's async despite the fact it's just a sync function reading something off the device. Even though the whole flow for this function is all synchronous and it *works* this way I think we should keep following the promise pattern. Thoughts?

File: packages/plugin-conversation/test/integration/spec/verbs.js [hunk 2]
```diff
@@ -51,6 +58,83 @@ describe(`plugin-conversation`, function() {
...
+      it(`removes the current user from the specified conversation`, () => spark.conversation.leave(conversation)
+        .then((activity) => {
+          assert.isActivity(activity);
+          return assert.isRejected(spark.conversation.get(conversation));
+        })
+        .then((reason) => {
+          assert.statusCode(reason, 404);
+          return assert.isRejected(spark.encryption.kms.fetchKey({uri: conversation.defaultActivityEncryptionKeyUrl}));
+        })
+        .then((reason) => assert.equal(reason.status, 403)));
```
Comment (Josh-Dykstra) [inline packages/plugin-conversation/test/integration/spec/verbs.js]: Any reason not to use `assert.statusCode` like above?

Comment (ianwremmel) [inline packages/plugin-conversation/test/integration/spec/verbs.js]: It didn't exist the first time this test was written

File: packages/plugin-conversation/src/conversation.js [hunk 3]
```diff
@@ -136,9 +181,10 @@ const Conversation = SparkPlugin.extend({
   },
 
   prepareConversation(conversation) {
-    return defaults(pick(conversation, `id`, `url`, `objectType`), {
+    const c = defaults(pick(conversation, `id`, `url`, `objectType`, `kmsResourceObjectUrl`, `defaultActivityEncryptionKeyUrl`), {
       objectType: `conversation`
     });
+    return c;
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/conversation.js]: yep.

File: packages/plugin-conversation/src/conversation.js (hunk 1 above)
Comment (ianwremmel) [inline packages/plugin-conversation/src/conversation.js]: it's because getServiceUrl returns a promise,

Comment (Josh-Dykstra) [inline packages/plugin-conversation/src/conversation.js]: I guess my comment was a long drawn out way of saying - why aren't we calling treating this call to _inferConversationUrl like every other one and handling the promise it returns?

Comment (ianwremmel) [inline packages/plugin-conversation/src/conversation.js]: oh. yea. that's a bug

**Pull request merged**